osu.Game/Skinning/Editor/SkinEditor.cs

```diff
@@ -330,6 +352,68 @@ private void revert()
                 // add back default components
                 getTarget(t.Target).Reload();
             }
+
+            RemoveMenuBackground();
+        }
+
+        private void removeMenuBackground()
+        {
+            RemoveMenuBackground();
+
+            onScreenDisplay?.Display(new SkinEditorToast(ToastStrings.MenuBackgroundChanged, CommonStrings.Default));
+        }
+
+        public void SetMenuBackground(FileInfo fileInfo)
+        {
+            using MemoryStream menuBackgroundStream = new MemoryStream();
+            using (var stream = fileInfo.OpenRead())
+                stream.CopyTo(menuBackgroundStream);
+            menuBackgroundStream.Seek(0, SeekOrigin.Begin);
+
+            Texture previewTexture = Texture.FromStream((skins as IO.IStorageResourceProvider).Renderer, menuBackgroundStream);
+            menuBackgroundStream.Seek(0, SeekOrigin.Begin);
+
+            currentSkin.Value.SkinInfo.PerformWrite(skinInfo =>
+            {
+                using (menuBackgroundStream)
+                    skins.AddFile(skinInfo, menuBackgroundStream, $"menu-background{fileInfo.Extension}");
+            });
+            realm.Run(r => r.Refresh());
```

Comment:
What is the purpose of this refresh? I believe that non-async realm writes already implicitly trigger refreshes?